IOS/SwiftSeedProject/SwiftSeedProject/Infrastructure/Network/Endpoint.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  Endpoint.swift
+//  SwiftSeedProject
+//
+//  Created by Brian Sztamfater on 3/4/17.
+//  Copyright © 2017 Making Sense. All rights reserved.
+//
+
+import Foundation
+import Alamofire
+
+enum Endpoint {
+    
+    case GetUser(userId: String)
+    case UpdateUser(userId: String)
+    case CreateUser()
+    
+    // MARK: - Public Properties
+    var method: Alamofire.HTTPMethod {
+        switch self {
+            case .GetUser:
+                return .get
+            case .UpdateUser:
+                return .put
+            case .CreateUser:
+                return .post
+        }
+    }
+    
+    var url: URL {
+        let baseUrl = NSURL.getBaseUrl()
+        switch self {
+            case .GetUser(let userId):
+                return baseUrl.appendingPathComponent("users/\(userId)")!
+            case .UpdateUser(let userId):
+                return baseUrl.appendingPathComponent("users/\(userId)")!
+            case .CreateUser():
+                return baseUrl.appendingPathComponent("users")!
+        }
+    }
+}
+
+private extension NSURL {
+    static func getBaseUrl() -> NSURL {
+        guard let info = Bundle.main.infoDictionary,
+            let urlString = info["Base URL"] as? String,
```

Comment:
Similar here, this implementation detail is hidden and could be annoying.

- Hidden because I will find out I have to define the base Url on the plist once I run a request and get this error. Although a developer can detect earlier that he have not provided a base Url anywhere and will have to look through the code and find this method (or give up and find the error later). This can be easily avoided by adding a explicit dependency somewhere so the first thing you have to do before instantiating whatever class is the initial point to this REST client design is to provide a base url. Then is up to the consumer if he want to retrieve that from the plist or anywhere else.

- There's no way to consume two different APIs within the same app with the current approach.